### PR TITLE
Fixup for #22510

### PR DESCRIPTION
**Pull request by tbu-**
r? @Manishearth

### CODE REVIEW

**Comment by huonw** [comment]
Could you adjust the commit message so that it stands alone more (e.g. "Revert incorrect `usize` -> `u32` replacements from #22510")?

File: src/libstd/thread.rs [hunk 1]
```diff
@@ -283,7 +283,7 @@ impl Builder {
         // address at which our stack started).
         let main = move || {
             let something_around_the_top_of_the_stack = 1;
-            let addr = &something_around_the_top_of_the_stack as *const isize;
+            let addr = &something_around_the_top_of_the_stack as *const i32;
```
Comment (Manishearth) [inline src/libstd/thread.rs]: Why change this?

**Comment by Manishearth** [comment]
Just need a small explanation there to confirm; though I suspect I understand what's going on. Also, agree with huon, a standalone message would be nice. Thanks!

File: src/libstd/thread.rs (hunk 1 above)
Comment (tbu-) [inline src/libstd/thread.rs]: It's irrelevant what `something_around_the_top_of_the_stack` is, so `i32` is just okay.

**Comment by tbu-** [comment]
@Manishearth Reworded the commit message as suggested.

**Comment by Manishearth** [comment]
This already merged

https://github.com/rust-lang/rust/commit/415c42ecc4aa425d71821b5807569937aca3128c

**Pull request closed without merging**